File: hannuota.c
#include <stdio.h>
int g_num = 0;

void hanoi(int n,char A,char B, char C){
	if(n == 1){
		printf("Move sheet %d from %c to %c\n",n,A,C);
		g_num += 1;
	}
	else{
		hanoi(n-1,A,C,B);
		printf("Move sheet %d from %c to %c\n", n, A, C);
		g_num += 1;
		hanoi(n-1,B,A,C);
	}
}

int main(){
	int n;
	printf("请输入盘数:");
	scanf("%d", &n);
	hanoi(n, 'A', 'B', 'C');
	printf("共计 %d 步\n", g_num);
	return 0;
}

File: hannuota.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-
# By who 2013-11-12

# TODO  hannuota.py


import sys
reload(sys)
sys.setdefaultencoding('utf-8')

g_count = 0

def hanoi(n, a, b, c):
	global g_count
	if n == 1:
		print("Move sheet %d from %s to %s"%(n, a, c))
		g_count += 1
	else:
		hanoi(n-1,a,c,b)
		print("Move sheet %d from %s to %s"%(n, a, c))
		g_count += 1
		hanoi(n-1,b,a,c)

def main():
	if len(sys.argv) <2:
		print "Please give me a number as the number of pan."
		return
	num = sys.argv[1]
	hanoi(int(num), 'A', 'B', 'C')
	print "Total : %d"%g_count

if __name__=="__main__":
	main()
